src/Http/Client.php

```diff
@@ -92,11 +94,29 @@ private function getCurlHandle(string $fullUrl, array $headers = []): CurlHandle
         curl_setopt($curlHandle, CURLOPT_ENCODING, '');
         curl_setopt($curlHandle, CURLOPT_HEADER, true);
 
-        if ($this->maxRedirects > 0) {
+        if ($maxRedirects > 0) {
             curl_setopt($curlHandle, CURLOPT_FOLLOWLOCATION, true);
-            curl_setopt($curlHandle, CURLOPT_MAXREDIRS, $this->maxRedirects);
+            curl_setopt($curlHandle, CURLOPT_MAXREDIRS, $maxRedirects);
         }
 
         return $curlHandle;
     }
+
+    private function parseRawHeaders(string $rawHeaders): array
+    {
+        $headers = explode("\n", $rawHeaders);
+        $headersArr = [];
+
+        foreach ($headers as $header) {
+            if (!str_contains($header, ':')) {
+                continue;
+            }
+
+            [$name, $value] = explode(':', $header, 2);
+
+            $headersArr[str_replace('-', '_', strtolower($name))] = trim($value);
```

Comment:
[I think](https://github.com/RogierW/rw-acme-client/pull/35#issuecomment-1820880231) this makes more sense:
```suggestion
            $headersArr[str_replace('_', '-', strtolower($name))] = trim($value);
```